backend/src/chat_server/utils.ts

```diff
@@ -33,3 +34,13 @@ export async function isBanned(username: string) {
     (!userState.banExpiration || userState.banExpiration > Date.now() / 1000)
   );
 }
+
+export function filterMessage(message:string) {
+  const badwords = BadWordsData.bad_words;
+  for (var i = 0; i < badwords.length; i++) {
+    var get = badwords[i].slice(0, -1).replace(/([a-z])/g, "$1[^a-z]*") + badwords[i].slice(-1);
+    var rxp = new RegExp(get, "ig");
+    message = message.replace(rxp, "****");
+  }
+  return message;
```

Comment:
I could be missing a lot of edge cases, but why couldn't we just write something like this?
```suggestion
  BadWordsData.bad_words.map(w => w.trim().toLowerCase()).forEach(w => message = message.replaceAll(w.toLowerCase(), "*".repeat(w.length()))
  return message;
```

Reply:
I've ran your solution and it have worked well. Just finished my lunch so I'll start with Jest right now.

Reply:
You right, your approach is much more readable and easier to understand by the way :+1: But I'll try to firgue out if there is any benifits of using my approach

Reply:
I think you can incoperate ur regex with what I've suggested.

In stead of replacing the string, build a regexp like what you did up there.

Reply:
Hmm, I don't really understand, if I build a regexp I still need to replace the string to ***. Would you mind explain it more clearly?

Reply:
<img width="1411" alt="Screen Shot 2020-09-06 at 5 17 09 PM" src="https://user-images.githubusercontent.com/5696959/92338402-01c0d800-f065-11ea-9652-6d0b87974f7b.png">

```javascript
BadWordsData.bad_words.map(w => [new RegExp(w.trim().split("").join("[^a-z]*"), "ig"), w.length]).forEach(wordRegexAndLength => message = message.replace(wordRegexAndLength[0], "*".repeat(wordRegexAndLength[1])));
```

Reply:
Of course break that code up, put comments for readability, but I think that should work. Let me know if there is any issue.

Reply:
![image](https://user-images.githubusercontent.com/8294297/92340505-d2ea4680-f0e4-11ea-85e4-6070898a5973.png)
I've tested with Jest and all passed. I've searched Google about lint and there is library for that, however, if we have any installed library, that would be great!